src/components/Firebase/firebase.js

```diff
@@ -79,5 +95,47 @@ class Firebase {
             }
         })
     }
+    addItem(state, type) {
+        const { title, quantity } = state
+        if (title == null || title === "") {
+            console.log("Title is null or empty")
+            return
+        }
+        if (type !== "Chores" && quantity == null) {
+            return
+        }
+        var firebaseFriendlyItem, historyItem;
+
+        switch (type) {
+            case "Chores":
+                var chore = new Chore(state.title, state.choreDeadlineDate, state.choreUsers, false, null);
+                firebaseFriendlyItem = chore.toFirestore()
+                historyItem = chore.toHistory(this.auth.currentUser.uid, false)
+                break;
+            case "Supplies":
+                var supply = new Supply(state.title, state.supplyQuantity, state.supplyDescription, false, null, null);
+                firebaseFriendlyItem = supply.toFirestore()
+                historyItem = supply.toHistory(this.auth.currentUser.uid, false)
+                break
+            case "Payments":
+                var payment = new Payment(state.title, state.paymentDeadlineDate, state.paymentQuantity, state.paymentUsers, state.paymentDescription, false, null, null)
+                firebaseFriendlyItem = payment.toFirestore()
+                historyItem = payment.toHistory(this.auth.currentUser.uid, false)
+                break
+            default:
+                break;
+        }
```

Comment:
Comments on this switch statement too. Is this part of loading from the database?

Reply:
I'll comment that as well, but for clarity that set's up the appropriate item model then turns it into something firebase can store. That part doesn't have any actual connection for firebase, it just sets it up

Reply:
Added some comments, do they help here?

Reply:
Commented that as well, let me know if it helped.